apps/wallet/src/ui/app/components/transactions-card/TxnIcon.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useMemo } from 'react';
+
+import Icon, { SuiIcons } from '_components/icon';
+
+import type { TransactionKindName } from '@mysten/sui.js';
+
+interface TxnItemIconProps {
+    txnKindName: TransactionKindName | 'Minted';
+    txnFailed?: boolean;
+    isSender: boolean;
+}
+
+export function TxnIcon({
+    txnKindName,
+    txnFailed,
+    isSender,
+}: TxnItemIconProps) {
+    const variant = useMemo(() => {
+        if (txnKindName === 'Minted') return 'Minted';
+        return isSender ? 'Send' : 'Receive';
+    }, [isSender, txnKindName]);
+
+    const icons = {
```

Comment:
You can hoist this out of the component.

Reply:
done